soh/src/overlays/actors/ovl_En_Dns/z_en_dns.c

```diff
@@ -164,7 +165,25 @@ void EnDns_Init(Actor* thisx, GlobalContext* globalCtx) {
     this->actor.speedXZ = 0.0f;
     this->actor.velocity.y = 0.0f;
     this->actor.gravity = -1.0f;
+    this->actor.textId = D_809F040C[this->actor.params];
     this->dnsItemEntry = sItemEntries[this->actor.params];
+    if (gSaveContext.n64ddFlag) {
+        // Ugly, but the best way we can identify which grotto we are in, same method 3DRando uses, but we'll need to account for entrance rando
```

Comment:
i know i did something different when identifying grottos for gossip stone hints in https://github.com/briaguya-ai/Shipwright/pull/106/files

any thoughts on how to make sure this works for entrance rando?

Reply:
The method used for grottos is grabbing the chest actor params, this won't work here as there isn't a chest in any of the scrub grottos.

For entrance randomizer we'll just need to make sure that the respawn data is set correctly when entering a scrub grotto. The respawn data will have to be set regardless for entrance rando, if not the grotto exits would be broken, it's just a matter of ensuring it's matched up with what's in the spoiler.